### DISCUSSION TITLE
STM32SD library reset due to SD reinsertion or random failure

### DISCUSSION THREAD

**Discussion by BrynMorgan1** [stm32duino/Arduino_Core_STM32#2348] [Libraries]
Hi there,

I'm using the STM32SD library with the STM32F411 on a custom PCB design. Our design is such that the user may remove and reinsert the SD card during operation. Additionally we occasionally observe random failures on the MicroSD side causing a lockup condition.
Due to these factors, we are looking to include a load switch for the MicroSD so it can be power cycled.

The STM32SD does not appear to have an end() function, and calling begin() a subsequent time appears to fail. Is there a way to disable, and reinitialise the STM32SD library to enable power cycle/reset of the MicroSD card?

Would usage of the BSP_SD_DeInit() function be suitable?

Kind regards,
Bryn

**Comment by fpistm** [comment]
Hi @BrynMorgan1 
When we implement the STM32SC we follows the Arduino SD library API and` end()` was not part of this API.
If you check the Arduino SD API reference:
https://www.arduino.cc/reference/en/libraries/sd/

`end()` is not reference.

Anyway it sees it was added:
https://github.com/arduino-libraries/SD/pull/37

Could you open an issue t the STM33SD GitHub repository to request the addition of this API. Of course, feel free to contribute. :wink:
